### PR TITLE
Cityscapes AutoLabelling dataset

### PR DESCRIPTION
**Pull request by lkdci**
Cityscapes AutoLabelled dataset were introduced by NVIDIA research group.
paper: Hierarchical Multi-Scale Attention for Semantic Segmentation", https://arxiv.org/abs/2005.10821
Official repo: https://github.com/NVIDIA/semantic-segmentation

This PR includes:
* `CityscapesConcatDataset` to support combination of cityscapes subsets.
* example ddrnet recipe with AL dataset.

### CODE REVIEW

**Review by Louis-Dupont — COMMENTED** [review]
Looks good, but let's also add a dataloader that would use this recipe + dataset. It's straightforward, just follow same pattern as with every other dataset:
https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/dataloaders/dataloaders.py

And then, if you can add a test for this dataloader, just to make sure that it runs
https://github.com/Deci-AI/super-gradients/blob/master/tests/unit_tests/dataloader_factory_test.py

Thanks 🙏

File: src/super_gradients/training/datasets/segmentation_datasets/cityscape_segmentation.py [hunk 1]
```diff
@@ -148,3 +148,32 @@ def target_transform(target):
...
+    >>> )
+
+    i.e to initiate a combination of the train-set with AutoLabelling-set:
+    >>> train_al_set = CityscapesConcatDataset(
+    >>>    root_dir='/data', list_files=['lists/train.lst', 'lists/auto_labelling.lst'], labels_csv_path='lists/labels.csv', ...
+    >>> )
+    """
+
+    def __init__(self, root_dir: str, list_files: List[str], labels_csv_path: str, **kwargs):
+        super().__init__(
```
[the remark this answers is not in the document]
Comment (lkdci) [reply src/super_gradients/training/datasets/segmentation_datasets/cityscape_segmentation.py]: Added [docs](https://github.com/Deci-AI/super-gradients/pull/1000/commits/9539dd7a2ee7a159b7460d51ec3d3fc77dbf220e)

**Review by Louis-Dupont — APPROVED** [review]
LGTM
